apps/src/netsim/netsimNodeFactory.js

```diff
@@ -0,0 +1,41 @@
+/* jshint
+ funcscope: true,
+ newcap: true,
+ nonew: true,
+ shadow: false,
+ unused: true,
+
+ maxlen: 90,
+ maxparams: 3,
+ maxstatements: 200
+ */
+'use strict';
+
+var netsimConstants = require('./netsimConstants');
+var NetSimClientNode = require('./NetSimClientNode');
+var NetSimRouterNode = require('./NetSimRouterNode');
+
+var NodeType = netsimConstants.NodeType;
+
+var netsimNodeFactory = module.exports;
+
+/**
+ * Given a set of rows from the node table on a shard, gives back a set of node
+ * controllers (of appropriate types).
+ * @param {!NetSimShard} shard
+ * @param {!Array.<Object>} rows
+ * @throws when a row doesn't have a mappable node type.
+ * @return {Array.<NetSimNode>} nodes for the rows
+ */
+netsimNodeFactory.nodesFromRows = function (shard, rows) {
```

Comment:
Broke this method into its own files because I hit a circular dependency between `NetSimRouterNode.js` and `netsimUtils.js`.  Wanted utils to be a low-dependency file and this seemed like a 'factory' behavior, so it moved here.